test/box-luatest/session_test.lua

```diff
@@ -0,0 +1,311 @@
+                      'type of trigger on_disconnect, no args')
+
+        local _, err = pcall(session.on_connect, function() end, function() end)
+        t.assert_str_contains(err, 'trigger reset: Trigger is not found',
+                              'on_connect trigger not found')
+        _, err = pcall(session.on_disconnect, function() end, function() end)
+        t.assert_str_contains(err, 'trigger reset: Trigger is not found',
+                              'on_disconnect trigger not found')
+
+        _, err = pcall(session.on_connect, 1, 2)
+        t.assert_str_contains(err, 'trigger reset: incorrect arguments',
+                              'on_connect bad args #1')
+        _, err = pcall(session.on_disconnect, 1, 2)
+        t.assert_str_contains(err, 'trigger reset: incorrect arguments',
+                              'on_disconnect bad args #1')
+
+        _, err = pcall(session.on_connect, 1)
+        t.assert_str_contains(err, 'trigger reset: incorrect arguments',
+                              'on_connect bad args #2')
+        _, err = pcall(session.on_disconnect, 1)
+        t.assert_str_contains(err, 'trigger reset: incorrect arguments',
+                              'on_disconnect bad args #2')
+
+        -- Use `nil` to clear the trigger.
+        session.on_connect(nil, fail)
+        session.on_disconnect(nil, fail)
+    end)
+end
+
+-- Check how connect/disconnect triggers work.
+g.test_session_triggers = function(cg)
+    cg.server:exec(function()
+        local session = box.session
+        local active_connections = 0
+        local function inc()
+            active_connections = active_connections + 1
+        end
+        local function dec()
+            active_connections = active_connections - 1
+        end
+        local net = { box = require('net.box') }
+        t.assert_type(session.on_connect(inc), 'function',
+                      'type of trigger inc on_connect')
+        t.assert_type(session.on_disconnect(dec), 'function',
+                      'type of trigger dec on_disconnect')
+        local fiber = require('fiber')
+        local uri = box.cfg.listen
+        local c = net.box.connect(uri)
+        while active_connections < 1 do fiber.sleep(0.001) end
+        t.assert_equals(active_connections, 1,
+                        'active_connections after 1 connection')
+        local net_box = require('net.box')
+        local c1 = net_box.connect(uri)
+        while active_connections < 2 do fiber.sleep(0.001) end
+        t.assert_equals(active_connections, 2,
+                        'active_connections after 2 connection')
+        c:close()
+        c1:close()
+        while active_connections > 0 do fiber.sleep(0.001) end
+        t.assert_equals(active_connections, 0,
+                        'active_connections after closing')
+
+        session.on_connect(nil, inc)
+        session.on_disconnect(nil, dec)
+
+        -- Write audit trail of connect/disconnect into a space.
```

Comment:
Please add a comment at the beginning that we check active connections for the whole test group scope. Otherwise, it looks like we should move the tests for the audit in the separate test group.

Reply:
Updated:

```diff
 
 -- Check how connect/disconnect triggers work.
+-- Note, we check active connections for the whole test group
+-- scope.
 g.test_session_triggers = function(cg)
     cg.server:exec(function()
         local session = box.session
```